arch/posix/include/posix_soc_if.h

```diff
@@ -37,6 +37,11 @@ void posix_irq_full_unlock(void);
 int  posix_get_current_irq(void);
 /* irq_offload() from irq_offload.h must also be defined by the SOC or board */
 
+static inline bool z_arch_irq_unlocked(unsigned int key)
```

Comment:
const unsigned int key?

Reply:
how is that helpful? it's passed by value